src/olympia/editors/tests/test_commands.py

```diff
@@ -142,6 +142,72 @@ def test_fetch_candidates(self):
         assert len(qs) == 1
         assert qs[0] == self.version
 
+    def test_fetch_candidates_post_review(self):
+        self.create_switch('post-review')
+        # Add nominated add-on: it should be considered since post-review
+        # waffle is enabled.
+        self.version.update(nomination=self.days_ago(1))
+        new_addon = addon_factory(status=amo.STATUS_NOMINATED, file_kw={
+            'status': amo.STATUS_AWAITING_REVIEW,
+            'is_webextension': True})
+        new_addon_version = new_addon.versions.all()[0]
+        new_addon_version.update(nomination=self.days_ago(2))
+
+        # Add a bunch of add-ons in various states that should not be returned.
+        # Public add-on with no updates.
+        addon_factory(file_kw={'is_webextension': True})
+
+        # Non-extension with updates.
+        search_addon = addon_factory(type=amo.ADDON_SEARCH)
+        version_factory(addon=search_addon, file_kw={
+            'status': amo.STATUS_AWAITING_REVIEW,
+            'is_webextension': True})
+
+        # Disabled add-on with updates.
+        disabled_addon = addon_factory(disabled_by_user=True)
+        version_factory(addon=disabled_addon, file_kw={
+            'status': amo.STATUS_AWAITING_REVIEW,
+            'is_webextension': True})
+
+        # Add-on with deleted version.
+        addon_with_deleted_version = addon_factory()
+        deleted_version = version_factory(
+            addon=addon_with_deleted_version, file_kw={
+                'status': amo.STATUS_AWAITING_REVIEW,
+                'is_webextension': True})
+        deleted_version.delete()
+
+        # Add-on with a non-webextension update.
+        non_webext_addon = addon_factory()
+        version_factory(addon=non_webext_addon, file_kw={
+            'status': amo.STATUS_AWAITING_REVIEW})
+
+        # Add-on with 3 versions:
+        # - one webext, listed, public.
+        # - one non-listed webext version
+        # - one listed non-webext awaiting review.
+        complex_addon = addon_factory(file_kw={'is_webextension': True})
+        version_factory(
+            addon=complex_addon, channel=amo.RELEASE_CHANNEL_UNLISTED,
+            file_kw={'is_webextension': True})
+        version_factory(addon=complex_addon, file_kw={
+            'status': amo.STATUS_AWAITING_REVIEW})
+
+        # Finally, add a second file to self.version to test the distinct().
+        file_factory(
+            version=self.version, status=amo.STATUS_AWAITING_REVIEW,
+            is_webextension=True)
+
+        # Gather the candidates.
+        command = auto_approve.Command()
+        command.post_review = True
+        qs = command.fetch_candidates()
+
+        # Only self.version should be found, once.
```

Comment:
I don't understand the comment vs. the lines below.  There are two items in qs, not one.